tests/NexusRpc.Tests/Handlers/HandlerTests.cs

```diff
@@ -0,0 +1,335 @@
+#pragma warning disable CA1822 // Don't require members to be static
+
+namespace NexusRpc.Tests.Handlers;
+
+using System.Text;
+using System.Threading.Tasks;
+using NexusRpc.Handlers;
+using Xunit;
+using Xunit.Abstractions;
+
+public class HandlerTests : TestBase
+{
+    public HandlerTests(ITestOutputHelper output)
+        : base(output)
+    {
+    }
+
+    [NexusService]
+    public interface ISimpleService
+    {
+        [NexusOperation]
+        string SayHello(string name);
+    }
+
+    [NexusServiceHandler(typeof(ISimpleService))]
+    public class SimpleSyncService
+    {
+        [NexusOperationHandler]
+        public IOperationHandler<string, string> SayHello() =>
+            OperationHandler.Sync<string, string>((context, name) => $"Hello, {name}");
+    }
+
+    [Fact]
+    public async Task Handler_SyncResult_Works()
+    {
+        var handler = new Handler(
+            [ServiceHandlerInstance.FromInstance(new SimpleSyncService())],
```

Comment:
Why is this `ServiceHandlerInstance.FromInstance` required? Can we have the handler take `[new SimpleSyncService()]` directly?

Reply:
Handlers work with service instances not just one form (the sugared form). Users can also manually instantiate a service handler instance, see the `Handler_ManuallyCreated_Works`. This is also how Java works today. It is plenty reasonable to ask those creating this handler to convert the sugared form to the usable form instead of us providing overloads that do it for them, especially in cases like this where it's a collection that could be some of both.